pkg/controllers/vdb/upgrade.go

```diff
@@ -645,24 +645,32 @@ func (i *UpgradeManager) isSubclusterIdle(ctx context.Context, pfacts *PodFacts,
 	return res, nil
 }
 
-// closeAllSessions will run a query to close all active user sessions.
-func (i *UpgradeManager) closeAllSessions(ctx context.Context, pfacts *PodFacts) error {
-	pf, ok := pfacts.findFirstPodSorted(func(v *PodFact) bool {
-		return v.isPrimary && v.upNode
-	})
+// areAllConnectionsPaused will run a query to see the number of non-superuser connections that are active (not paused)
+// it returns a requeue error if there are still active connections
+func (i *UpgradeManager) areAllConnectionsPaused(ctx context.Context, pfacts *PodFacts) (bool, error) {
```

Comment:
```suggestion
func (i *UpgradeManager) areAllConnectionsPaused(ctx context.Context, pfacts *PodFacts) (ctrl.Result, error) {
```
I suggest you follow the `isSubclusterIdle()` logic.